titiler/pgstac/db.py

```diff
@@ -6,10 +6,8 @@
 
 from fastapi import FastAPI
 
-settings = PostgresSettings()
 
-
-async def connect_to_db(app: FastAPI) -> None:
+async def connect_to_db(app: FastAPI, settings: PostgresSettings) -> None:
     """Connect to Database."""
     app.state.dbpool = ConnectionPool(
```

Comment:
I guess we could make this change opt-in by doing something like this:

```suggestion
async def connect_to_db(app: FastAPI, settings: PostgresSettings = None) -> None:
    """Connect to Database."""
    if not settings:
        settings = PostgresSettings()
    app.state.dbpool = ConnectionPool(
```

Reply:
I'm not sure to understand, we still need to pass some settings to `ConnectionPool`

Reply:
Right, and that still happens after the changes in this PR. The key point is to not run `PostgresSettings()` for the user. We don't want to do because different users may have different techniques for getting the settings (eg from `.env` file, from a secrets manager, or hard-coded in their codebase).  Instead we should give the user the opportunity to pass on the settings after they initialize the settings however they see fit. 

The above code suggestion would allow the user to pass in the settings manually, or if not provided then Titiler will attempt to pull it out of the environment (just as it works today).

Reply:
Here's an example of how we would use this:


```py
# config.py
import boto3
from titiler.pgstac.settings import PostgresSettings as PostgresBaseSettings


class PostgresSettings(PostgresBaseSettings):
    @classmethod
    def from_secret(cls, secretsmanager_arn: str) -> "PostgresSettings":
        client = boto3.client("secretsmanager")
        secret = client.get_secret(arn=secretsmanager_arn)  # TODO: This is dummy code
        return cls.parse_obj(
            {
                "postgres_user": secret["username"],
                "postgres_pass": secret["password"],
                "postgres_host": secret["host"],
                "postgres_port": secret["port"],
                "postgres_dbname": secret["dbname"],
            }
        )
```

```py
# main.py
import os

from .config import PostgresSettings

pg_settings = (
    PostgresSettings.from_secret(secret_arn)
    if (secret_arn := os.environ.get("PGSTAC_SECRET_ARN"))
    else PostgresSettings()
)

# ... define app, etc

@app.on_event("startup")
async def startup_event() -> None:
    """Connect to database on startup."""
    await connect_to_db(app, settings=pg_settings)
```